stocks/stocks.py

```diff
@@ -1,7 +1,155 @@
+    else:
+        cur.execute("SELECT * FROM stock_tracker")
+        tracked_stocks = cur.fetchall()
+
+    if any(stock is None for stock in tracked_stocks):
+        return 
+
+    tracked_stocks_list = []
+    for stock_details in tracked_stocks:
+        cur.execute("SELECT symbol, name FROM stock WHERE id=?", [stock_details[6]]) 
+        stock_profile = cur.fetchone()
+        symbol = stock_profile[0]
+        name = stock_profile[1]
+        average_cost = stock_details[1]
+        percent = stock_details[2]
+        increase = True if stock_details[3] else False
+        decrease = True if stock_details[4] else False
+        last_modified = stock_details[5]
+
+        if detailed:
+            response = get_stock_quote(symbol)
+            if not response:
+                return jsonify({"Error": "You are tracking an invalid stock symbol: {symbol}".format(symbol=symbol)}), 404        
+            percent_difference = calculate_percent_change(response, average_cost)
+            tracked_stocks_list.append({"symbol": symbol, "name": name, "percent_difference": percent_difference,
+                                        "last_modified": last_modified, "alert_on_increase": increase, "alert_on_decrease": decrease, 
+                                        "average_cost": average_cost, "percent_to_track_threshold": percent})
+        else:
+            tracked_stocks_list.append({"symbol": symbol, "name": name})
+
+    return tracked_stocks_list
+
+
+def trigger_alert(stocks_increased, stocks_decreased):
+    """
+    Trigger alert given tracked stocks that increased or decreased
+    """
+    increase_alert_message = ""
+    decrease_alert_message = ""
+
+    if stocks_increased:    
+        increase_alert_message = "Stocks increased: \n"
+        for stock in stocks_increased:
+            increase_alert_message = increase_alert_message + "[" + stock.get("symbol") + "]" + stock.get("name") + " went up " + u'\u2191' + str(stock.get("percent_increase")) + "%\n"
+
+    if stocks_decreased:
+        decrease_alert_message = "Stocks decreased: \n"
+        for stock in stocks_decreased:
+            decrease_alert_message = decrease_alert_message + "[" + stock.get("symbol") + "]" + stock.get("name") + " went down " + u'\u2193' + str(stock.get("percent_decrease")) + "%\n"
+    
+    return increase_alert_message + "\n" + decrease_alert_message 
+
+
+def get_tracked_stocks():
+    """
+    Run caculation of percent change for each tracked stock
+    Trigger alert if increase or decrease is enabled and percent threshold is met
+    Cron will call this function
+    """
+    tracked_stocks_list = filter_tracked_stocks_details(None, True)
+    
+    stocks_increased = []
+    stocks_decreased = []
+    for stock_detail in tracked_stocks_list:
+        if stock_detail.get("alert_on_increase") and stock_detail.get("percent_difference") >= stock_detail.get("percent_to_track_threshold") and stock_detail.get("percent_difference") > 0:
```

Comment:
This can be a little hard to read. You could break it down in a few ways, for example:
```
some_var = stock_detail.get("alert_on_increase") and stock_detail.get("percent_difference")
some_var_2 = stock_detail.get("percent_to_track_threshold") and stock_detail.get("percent_difference") > 0

if some_var_1 >= some_var_2:
    # do something
```